cmd/guacone/cmd/s3.go

```diff
@@ -49,8 +50,29 @@ type s3Options struct {
 
 var s3Cmd = &cobra.Command{
 	Use:   "s3 [flags]",
-	Short: "listens to kafka/sqs s3 events to download documents and add them to the GUAC graph, or directly downloads from s3",
-	Args:  cobra.MinimumNArgs(0),
+	Short: "takes SBOMs and attestations from S3 compatible bucket and injects them to GUAC graph",
+	Long: `S3 collector can download one item from the storage, the whole bucket or listen to storage events using sqs/kafka (poll) and download the files as they are uploaded.
```

Comment:
"the whole bucket or listen to storage events using sqs/kafka (poll)"

Is sqs or kafka needed in this case? Could we not do polling similar to the GCP bucket? It seems like it adds complexity.

Reply:
Could we keep rechecking the bucket but only grab items based on the creation or modification datetime?

Reply:
We can do that, but it's not how people use S3 service idiomatically. And it seems to me it will in the end add more complexity (handling failures and retries always end up the biggest part of the solution like that). 
I'm not sure why is the current approach considered too complex. It hides everything behind proper interfaces, so people can extend it for their needs. I understand it adds two more dependencies, but we already have a lot of dependencies in the project so it's not so overwhelming.
Without proper notification support, we definitely couldn't use S3 collector for our needs.

Reply:
I understand your point the queue method is much more robust. So, it's not about the dependencies, it's more about the de-duplication of code. I am working with https://gocloud.dev/ which provides abstraction for both the blob store and the pubsub. I am already working to integrate this with the rest of GUAC (allowing users to use any pubsub they want and not be tied to NATS). Similarly, providing the mechanism to add a blob store that can be used (in combination with CDEvents) to pass large documents more efficiently between the `guaccollect` and `guacingest`. Lets chat in an issue or offline and I can walk you through what I am thinking and if that makes sense to do here. I will not hold up this PR any longer :)